src/test/java/dog/pawbook/logic/parser/EditProgramCommandParserTest.java

```diff
@@ -0,0 +1,179 @@
+
+        // multiple invalid values, but only the first invalid value is captured
+        assertParseFailure(parser, "1" + INVALID_NAME_DESC + INVALID_SESSION_DESC + VALID_TAG_ALL,
+                Name.MESSAGE_CONSTRAINTS);
+    }
+
+    @Test
+    public void parse_allFieldsSpecified_success() {
+        Integer targetId = ID_TWO;
+        String userInput = targetId + TAG_DESC_ALL + SESSION_DESC_OBEDIENCE_TRAINING
+                + NAME_DESC_OBEDIENCE_TRAINING + TAG_DESC_PUPPIES;
+
+        EditProgramDescriptor descriptor = new EditProgramDescriptorBuilder().withName(VALID_NAME_OBEDIENCE_TRAINING)
+                .withSessions(VALID_SESSION_OBEDIENCE_TRAINING).withTags(VALID_TAG_ALL, VALID_TAG_PUPPIES).build();
+        EditProgramCommand expectedCommand = new EditProgramCommand(targetId, descriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_someFieldsSpecified_success() {
+        Integer targetId = ID_ONE;
+        String userInput = targetId + SESSION_DESC_OBEDIENCE_TRAINING + NAME_DESC_POTTY_TRAINING;
+
+        EditProgramDescriptor descriptor = new EditProgramDescriptorBuilder().withName(VALID_NAME_POTTY_TRAINING)
+                .withSessions(VALID_SESSION_OBEDIENCE_TRAINING).build();
+        EditProgramCommand expectedCommand = new EditProgramCommand(targetId, descriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_oneFieldSpecified_success() {
+        // name
+        Integer targetId = ID_THREE;
+        String userInput = targetId + NAME_DESC_OBEDIENCE_TRAINING;
+        EditProgramDescriptor descriptor = new EditProgramDescriptorBuilder()
+                .withName(VALID_NAME_OBEDIENCE_TRAINING).build();
+        EditProgramCommand expectedCommand = new EditProgramCommand(targetId, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // sessions
+        userInput = targetId + SESSION_DESC_OBEDIENCE_TRAINING;
+        descriptor = new EditProgramDescriptorBuilder().withSessions(VALID_SESSION_OBEDIENCE_TRAINING).build();
+        expectedCommand = new EditProgramCommand(targetId, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // tags
+        userInput = targetId + TAG_DESC_ALL;
+        descriptor = new EditProgramDescriptorBuilder().withTags(VALID_TAG_ALL).build();
+        expectedCommand = new EditProgramCommand(targetId, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_multipleRepeatedFields_acceptsLast() {
+        Integer targetId = ID_ONE;
+        String userInput = targetId + TAG_DESC_PUPPIES + SESSION_DESC_OBEDIENCE_TRAINING + TAG_DESC_ALL
+                + TAG_DESC_ALL + TAG_DESC_PUPPIES + SESSION_DESC_POTTY_TRAINING
+                + SESSION_DESC_POTTY_TRAINING + TAG_DESC_PUPPIES + SESSION_DESC_OBEDIENCE_TRAINING;
```

Comment:
the method name implies that multiple fields that should only be specified once are now specified multiple times, in which only the last repetition is to be accepted as the final value, programs clearly don't have such a field when being edited, both tags and sessions accepts all of the fields and will only ignore repeated values, what I need from you is to write test methods that instead test if multiple tags & sessions are all accepted.

Reply:
> what I need from you is to write test methods that instead test if multiple tags & sessions are all accepted.
You mean to test multiple tags and sessions?

Reply:
Yes, multiple valid but different tags and sessions, ensure that not only the last is accepted.

Reply:
Done! Tell me if it's still not enough

Reply:
> what I need from you is to write test methods that instead test if multiple tags & sessions are all accepted

This refers to additional test methods, the current test method in which you added additional stuff is pointless, keep it as the original is fine. Write **another** test method that specifically test multiple valid and different tags overriding the existing ones, and another that test multiple valid and different session timings to see if it overrides the existing ones